aiosmtpd/smtp.py

```diff
@@ -591,6 +593,32 @@ def _syntax_available(self, method):
         status = await self._call_handler_hook('RSET')
         await self.push('250 OK' if status is MISSING else status)
 
+    def _parse_xclient_attributes(self, attributes):
+        """
+        Parse key-value arguments and return OrderedDict on success.
+        Raise ValueError on wrong arguments.
+        """
+        result = collections.OrderedDict()
+        for kv in attributes.split(' '):
+            key, value = kv.split('=')
+            if key not in XCLIENT_ATTRIBUTES:
+                raise ValueError()
```

Comment:
You don't need to instantiate the exception here, unless you're going to give the instance additional details.  That might actually be useful though: should the bogus `key` be included in the exception message?